### PR TITLE
Flattening telemetry "SHARE" Events

### PR DESCRIPTION
**Pull request by manjudr**
1. Routing all the "SHARE" Events to the new kafka topic (`telemetry.share`) in the telemetry router samza job.
2. Events flattener listen to `telemetry.share` kafka topic to flatten the share events and push all flaten and share events  into `telemetry.sink` kafka topic.

**Code Coverage: 100%**
Screenshot 2020-01-10 at 4 39 25 PM

### CODE REVIEW

File: data-pipeline/events-flattener/src/main/java/org/ekstep/ep/samza/service/EventsFlattenerService.java [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+
+    /**
+     * @param source
+     * @param sink
+     */
+    public void process(EventsFlattenerSource source, EventsFlattenerSink sink) {
+        Event event = source.getEvent();
+        if (event.eid().equals(config.getFlattenEventName())) {
+            try {
+                this.toFlatten(event, getClonedEventObject(event), sink);
```
Comment (SanthoshVasabhaktula) [inline data-pipeline/events-flattener/src/main/java/org/ekstep/ep/samza/service/EventsFlattenerService.java]: Dispatch the original event to the sink topic. We can avoid the cloning...

File: data-pipeline/events-flattener/src/main/java/org/ekstep/ep/samza/task/EventsFlattenerConfig.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
+package org.ekstep.ep.samza.task;
+
+import org.apache.samza.config.Config;
+
+public class EventsFlattenerConfig {
+
+    private final String JOB_NAME = "Events Flattener";
```
Comment (SanthoshVasabhaktula) [inline data-pipeline/events-flattener/src/main/java/org/ekstep/ep/samza/task/EventsFlattenerConfig.java]: Job name should be Capital case without spaces

File: data-pipeline/events-flattener/src/main/java/org/ekstep/ep/samza/service/EventsFlattenerService.java [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+                        if (Double.parseDouble(transfers) > 0) {
+                            eDataType = IMPORTED_KEY;
+                        }
+                        clonedEvent.updatedEventEdata(eDataType, Double.parseDouble(p.get("size")));
+                    } else {
+                        clonedEvent.updatedEventEdata(orginalEvent.edataType(), null);
+                    }
+                }
+            } else {
+                clonedEvent.updatedEventEdata(orginalEvent.edataType(), null);
```
Comment (SanthoshVasabhaktula) [inline data-pipeline/events-flattener/src/main/java/org/ekstep/ep/samza/service/EventsFlattenerService.java]: Original object edata attributes "dir" and "type" should also flow into the edata of each flattened event

**Comment by manjudr** [comment]
Screenshot 2020-01-10 at 4 39 25 PM

**Code coverage improved to 100%**

**Pull request merged**